product-mini/platforms/linux-sgx/enclave-sample/Makefile

```diff
@@ -102,6 +102,7 @@ Enclave_Include_Paths := -IEnclave -I$(WAMR_ROOT)/core/iwasm/include \
 Enclave_C_Flags := $(SGX_COMMON_CFLAGS) -nostdinc -fvisibility=hidden -fpie -fstack-protector $(Enclave_Include_Paths)
 ifeq ($(SPEC_TEST), 1)
 	Enclave_C_Flags += -DWASM_ENABLE_SPEC_TEST=1
+	Enclave_C_Flags += -DSGX_DISABLE_WASI
```

Comment:
Don't disable WASI by default, SGX supports WASI and enables WASI by default.

Reply:
Since SPEC test cases are all from WAST and can't be compiled with `wasi-sdk-clang` and they are not using any WASI apis. we'd better run spec test cases by avoiding WASI module rules.

Reply:
That's because that  we didn't check module->is_wasi_mode when calling `_start` and `_initialize`. My understanding is that we check the wasi module rules only when the module is wasi moduel, or many non wasi cases may run failed.

Reply:
The point here is we should run SPEC-TEST in **non-wasi-mode**.  the modification won't change the default setting of SGX.